From b0a1838e3c638409c1c2b7c3f54444c790211e5c Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 13 Mar 2024 12:35:01 +0200
Subject: [PATCH] [FSW-402] fs1: bootrom_init: read cold rst reg data

This patch adds functionality to read cold rst reg data for
IO die master die.

Change-Id: I4029239e52f58a71a300fb01df298a1c19b8127f
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index aff11a8..11bd902 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -95,3 +95,13 @@
 {
 	return IO0_HALF0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_SW_STATUS_0);
 }
+
+void boot_rom_read_cold_flops(void)
+{
+	struct fs1_cold_rst_data cold_rst_data;
+
+	cold_rst_data.data = regmap_readl(IO0_HALF0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+	boot_rom_global.recovery_flag = cold_rst_data.recovery_flag;
+	boot_rom_global.validation_pass = cold_rst_data.validation_flag;
+	boot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
+}
diff --git a/common/common.c b/common/common.c
index 568d9a9..849e246 100644
--- a/common/common.c
+++ b/common/common.c
@@ -24,5 +24,10 @@
 
 	boot_rom_status_update(PROGRESS_INIT);
 
-        boot_rom_init_ts();
+	boot_rom_init_ts();
+
+	if (boot_rom_global.die_type != IO_DIE_MASTER)
+		return;
+
+	boot_rom_read_cold_flops();
 }
diff --git a/common/common.h b/common/common.h
index 0944fd9..dfb5ed6 100644
--- a/common/common.h
+++ b/common/common.h
@@ -11,5 +11,6 @@
 
 void boot_rom_clear_status(void);
 void boot_rom_persistent_dccm_init(uint8_t die_id);
+void boot_rom_read_cold_flops(void);
 
 #endif /* _COMMON_H_ */
